packages/shell-api/src/decorators.ts

```diff
@@ -36,27 +36,81 @@ export interface ShellResultSourceInformation {
 }
 
 export interface ShellResult {
-  value: any;
-  type: string;
+  /// The original result of the evaluation, as it would be stored e.g. as a
+  /// variable inside the shell.
+  rawValue: any;
+
+  /// A version of the raw value that is usable for printing, e.g. what the
+  /// shell would print.
+  printable: any;
+
+  /// The type of the shell result. This refers to built-in shell types, e.g.
+  /// `Cursor`; all unknown object types and primitives are given the
+  /// type `null`.
+  type: string | null;
+
+  /// Optional information about the original data source of the result.
   source?: ShellResultSourceInformation;
+
+  /// @deprecated Use either `printable` or `rawValue`.
+  value: any;
 }
 
 export class ShellApiClass implements ShellApiInterface {
   help: any;
-  [asShellResult](): any {
+  get [shellApiType](): string {
```

Comment:
@aherlihy did we have issues with getters and setters in the past with he java shell?

Reply:
Exactly, you would have to update the java shell since they use this method directly (which is why it was the only non-symbol of these internal methods)

Reply:
Are you thinking about `_asPrintable()` here? That is still left as an alias for the java shell (`asShellResult` is not mentioned in their source, and always was a symbol)

Reply:
Yes I was! Sorry the comment was super unclear. I just meant that the reason _asPrintable was different from the others was because of the java shell